js/main.js

```diff
@@ -32,38 +33,65 @@ function saveEntryHandle(event) {
   var formDataObj = {
     title: $formElement.elements.title.value,
     imgURL: $formElement.elements.photo.value,
-    notes: $formElement.elements.notes.value,
-    entryId: data.nextEntryId
+    notes: $formElement.elements.notes.value
   };
-  data.entries.unshift(formDataObj);
-  data.nextEntryId++;
+  if (data.editing === null) {
+    formDataObj.entryId = data.nextEntryId;
+    data.entries.unshift(formDataObj);
+    data.nextEntryId++;
+    var $newEntry = renderEntry(formDataObj);
+    $entryViewList.prepend($newEntry);
+  } else {
+    formDataObj.entryId = data.editing.entryId;
+    for (var i = 0; i < data.entries.length; i++) {
+      if (data.entries[i].entryId === data.editing.entryId) {
+        data.entries[i].title = formDataObj.title;
+        data.entries[i].imgURL = formDataObj.imgURL;
+        data.entries[i].notes = formDataObj.notes;
+      }
+    }
+    var $editedEntry = renderEntry(formDataObj);
+    var $itemNodes = document.querySelectorAll('i');
+    for (var j = 0; j < $itemNodes.length; j++) {
+      if (parseInt($itemNodes[j].getAttribute('data-entry-id')) === formDataObj.entryId) {
+        var $itemToBeReplaced = $itemNodes[j].closest('.entry-list-item');
```

Comment:
I would `break` once you find the right item, so you don't do wasted work on subsequent iterations.